src/streams/streamsActions.js

```diff
@@ -19,13 +19,18 @@ export const updateExistingStream = (
     getZulipFeatureLevel(state) >= 64 ? value : JSON.stringify(value);
 
   const auth = getAuth(state);
+  const updatedAndEncodedValues = {};
   if (initialValues.name !== newValues.name) {
-    await api.updateStream(auth, id, 'new_name', maybeEncode(newValues.name));
+    updatedAndEncodedValues.new_name = maybeEncode(newValues.name);
```

Comment:
The "encode" part here isn't fundamental -- it's basically a mistake in the server API, one that's corrected in newer server versions, plus the dealing with it should ideally go inside the API binding code. So I'd leave it out of this variable's name, letting the name refer just to what this would ideally be, and let the TODO comments above cover the fact that there's this hack that makes it not quite what it would ideally be.

Perhaps "updates"?